### ISSUE TITLE
aws/asg: Wait for scaling activities before deleting

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/terraform#1512]
Sometimes I end up with and error trying to destroy an ASG:

```
You cannot delete an AutoScalingGroup while there are scaling activities in progress for that group.
```

I guess you can simulate that state by manually changing "desired count" in the web console and then immediately trigger `terraform destroy`.

I believe that Terraform should try at least couple times and wait some time until ASG finishes these scaling activities and then destroy the ASG.

**Comment by phinze** [comment]
I wonder if it's actually more appropriate for TF to actually pause and resume scaling operations when making changes to ASGs. That's what most bespoke tooling I've worked with does, and it seems pretty reasonable.

https://docs.aws.amazon.com/AutoScaling/latest/DeveloperGuide/US_SuspendResume.html

What do you think @radeksimko?

**Comment by radeksimko** [comment]
Hmmm, I didn't know it's actually possible, thanks for teaching me something new! :+1: :smiley: 
It is probably the best solution as long as the API call is synchronous.

i.e. if there's scaling activity in progress, it either wait until that activity finishes and then puts the lock in place that will prevent any further activities to occur OR if it just suspends whatever's happening (which would bring another question `->` if I'm bringing up a new instance during that autoscaling activity, what happens with that one? is it terminated?)

In simple terms - can that API call to "suspend" auto scaling processes fail?

**Comment by phinze** [comment]
Yeah good question. Going over the docs it seems like it's generally expected to succeed, which makes me feel like it's only going to suspend future operations. We'd have to do a little testing to see if the effect takes place immediately and what the edge cases might be.

**Comment by woodhull** [comment]
+1 calling suspend when Terraform operations are in progress on ASGs seems like the correct behavior.

**Comment by radeksimko** [comment]
Fixed via #1840
